xbmc/URL.cpp

```diff
@@ -253,35 +253,29 @@ void CURL::Parse(const std::string& strURL1)
     }
   }
 
+  std::string strHostNameAndPort = strURL.substr(iPos, (iSlash == std::string::npos) ? iEnd - iPos : iSlash - iPos);
+  // check for IPv6 numerical representation inside []
+  size_t iBrk = strHostNameAndPort.rfind("]");
+  if (iBrk < iEnd && iBrk != std::string::npos && strHostNameAndPort.find("[") == 0)
+  {
+    m_strHostName = strHostNameAndPort.substr(1, iBrk-1);
+    strHostNameAndPort.erase(0, iBrk+1);
```

Comment:
mhhh also here i am unsure if this is not prone to buffer overrun. Its hard to see where the bounds are garanteed here (in case its hidden in this if condition)